android/src/main/java/com/pdftron/pdftronflutter/FlutterDocumentActivity.java

```diff
@@ -72,29 +74,38 @@ public static void openDocument(Context packageContext, String document, String
         }
     }
 
-    public static void openDocument(Context packageContext, Uri fileUri, String password, @Nullable JSONObject customHeaders, @Nullable ViewerConfig config) {
+    public static void openDocument(Context packageContext, Uri fileUri, String password, @Nullable JSONObject customHeaders, ViewerConfig config) {
         openDocument(packageContext, fileUri, password, customHeaders, config, DEFAULT_NAV_ICON_ID);
     }
 
-    public static void openDocument(Context packageContext, Uri fileUri, String password, @Nullable JSONObject customHeaders, @Nullable ViewerConfig config, @DrawableRes int navIconId) {
-        DocumentActivity.IntentBuilder intentBuilder = DocumentActivity.IntentBuilder.fromActivityClass(packageContext, FlutterDocumentActivity.class);
+    public static void openDocument(Context packageContext, Uri fileUri, String password, @Nullable JSONObject customHeaders, ViewerConfig config, @DrawableRes int navIconId) {
 
-        if (null != fileUri) {
-            intentBuilder.withUri(fileUri);
-        }
+        if (sPdfViewCtrlTabHostFragment != null) {
```

Comment:
I think static PdfViewCtrlTabHostFragment2 is unnecessary. We can always obtain the fragment by calling getPdfViewCtrlTabHostFragment.

Instead of storing a static fragment in onCreate and clearing it onDestroy, just set a boolean flag.